startup/30-basler.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+import itertools
+
+import matplotlib.pyplot as plt
+import numpy as np
+from pypylon import pylon
+
+from ophyd import Component as Cpt
+from ophyd import Device, Signal
+from ophyd.sim import NullStatus
+
+import os
+
+os.environ['PYLON_CAMEMU'] = "1"
+
+plt.ion()
+
+class BaslerCamera(Device):
+
+    image = Cpt(Signal, kind="normal")
+    mean  = Cpt(Signal, kind="hinted")
+    shape = Cpt(Signal, kind="normal")
+
+    def __init__(self, name='basler_cam',
+                    sim_id=None, watch_name=None,
+                    sirepo_server='http://10.10.10.10:8000', source_simulation=False,
+                    root_dir='/tmp/sirepo_det_data', **kwargs):
+        super().__init__(name=name, **kwargs)
+
+        transport_layer_factory = pylon.TlFactory.GetInstance()
+        device_info_list = transport_layer_factory.EnumerateDevices()
+        print(device_info_list)
+
+        number_of_devices = len(device_info_list)
+        print(f"{number_of_devices = }")
+
+        self.device_info = device_info_list[0]
+        self.device = transport_layer_factory.CreateDevice(self.device_info)
+        self.camera_object = pylon.InstantCamera(self.device)
+
+        self.camera_object.Open()
+
+        self.user_defined_name = self.camera_object.GetDeviceInfo().GetUserDefinedName()
+        self.camera_model = self.camera_object.GetDeviceInfo().GetModelName()
+        self.camera_serial_no = self.camera_object.GetDeviceInfo().GetSerialNumber()
+        self.width = self.camera_object.Width()
+        self.height = self.camera_object.Height()
+        self.pixel_level_min = self.camera_object.PixelDynamicRangeMin()
+        self.pixel_level_max = self.camera_object.PixelDynamicRangeMax()
+        self.active_format = self.camera_object.PixelFormat.GetValue()
+        self.formats_supported = self.camera_object.PixelFormat.Symbolics
+        self.payload_size = self.camera_object.PayloadSize()
+        self.grab_timeout = 5000 # ms
+
+        self.camera_object.Close()
```

Comment:
I think these lines can be moved to the `stage` method. This way we don't "touch" the camera API on instantiation, but rather access the API on explicit use.